sqlx-postgres/src/options/mod.rs

```diff
@@ -218,6 +225,26 @@ impl PgConnectOptions {
         self
     }
 
+    /// Sets the protocol with which the secure SSL TCP/IP connection will be negotiated with
+    /// the server.
+    ///
+    /// By default, the protocol is [`Postgres`](PgSslNegotiation::Postgres), and the client will
+    /// first check whether the server supports SSL, and fallback to a non-SSL connection if not.
+    ///
+    /// Ignored for Unix domain socket communication.
```

Comment:
Is this the case? I don't think it is. We still attempt a TLS upgrade when connecting over Unix sockets: https://github.com/launchbadge/sqlx/blob/a301d9abad984690f69a4356a5da816d5bdc9319/sqlx-postgres/src/connection/stream.rs#L46

That's arguably unnecessary, it's just the existing behavior.

Reply:
IIRC I based this on the docstring for `ssl_mode`, but I think the same question applies there too.

I can't see anywhere that the `ssl_mode` configuration is ignored when using a Unix socket, so I think the line should probably be removed from both docstrings. Thoughts?

Reply:
Yeah, go ahead and remove that. Though I wonder if Postgres will even accept a TLS upgrade over a Unix domain socket.

Reply:
Please see latest push.